cli/tests/main_test.go

```diff
@@ -0,0 +1,80 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// TestMain builds the rudder-cli binary once, exposes it via PATH and then
+// runs the package tests. It honours the cli/logging and cli/testing rules by
+// printing only essential information and performing cleanup after execution.
+func TestMain(m *testing.M) {
+	tmpDir, err := os.MkdirTemp("", "rudder-cli-bin-*")
+	if err != nil {
+		fmt.Println("failed to create temp dir:", err)
+		os.Exit(1)
+	}
+
+	defer os.RemoveAll(tmpDir)
+
+	exec, err := NewCmdExecutor("")
+	if err != nil {
+		fmt.Println("failed to init executor:", err)
+		os.Exit(1)
+	}
+
+	bin, err := NewCLIBinary(tmpDir, exec)
+	if err != nil {
+		fmt.Println("failed to init CLIBinary helper:", err)
+		os.Exit(1)
+	}
+
+	path, err := bin.Setup()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	oldpath := os.Getenv("PATH")
+	binDir := filepath.Dir(path)
+
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", binDir, oldpath))
```

Comment:
Do we need to put the binary in the PATH? Can we just rely on the absolute path of the binary?